pkg/resources/pods.go

```diff
@@ -75,6 +75,19 @@ func (pc PodAccessor) PodIPsByAge() ([]string, error) {
 	if err != nil {
 		return nil, err
 	}
+	// Keep only running ones.
+	write := 0
+	for i := range pods {
+		p := pods[i]
+		if p.Status.Phase == corev1.PodRunning && p.DeletionTimestamp == nil {
+			if write != i {
```

Comment:
is this check needed? seems like you could probably just do pods[write] = p (and that’s how I’ve always seen this pattern elsewhere)

Reply:
(Eg https://github.com/golang/go/wiki/SliceTricks#filter-in-place has it without the check)